src/oc-id/app/controllers/profiles_controller.rb

```diff
@@ -118,4 +121,16 @@ def valid_signature?
       params[:email],
     ).valid_for?(params[:signature])
   end
+
+  def verify_user_params
+    @user = current_user
+    begin
+      raise TypeError, "Invalid First Name." unless Constants::REGEXP_NAME.match(params[:user][:first_name])
```

Comment:
Does the underlying /users API require the firstname and last name?  The spec here indicates that they are optional: https://github.com/chef/chef-server/blob/master/src/oc_erchef/apps/chef_objects/src/chef_user.erl#L190-L200  But I haven't read through all the code to see if they may be require elsewhere.